### PR TITLE
Add functionality for .tsx/.jsx files

### PR DESCRIPTION
**Pull request by dwalsh01**
As discussed in the raised issue, here is my pull request.

https://github.com/jparise/vim-graphql/issues/32

### CODE REVIEW

**Comment by kaicataldo** [comment]
Will this also work for `.jsx` files?

**Comment by dwalsh01** [comment]
@kaicataldo to the best of my knowledge yes, it worked when I tested it.

File: after/syntax/tsx.vim [hunk 1]
```diff
@@ -0,0 +1,31 @@
+"""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
+" Vim syntax file
+"
+" Language: TSX (TypeScript)
+"
+"""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
```
Comment (jparise) [inline after/syntax/tsx.vim]: Please remove this header block. We don't use this style elsewhere in the plugin code.

**Review by jparise — COMMENTED** [review]
Your change looks good. Can you also do these two things before I merge it?

1. Add a note to `README.md` about this new support.
2. Add a new unit test (to the `test/` directory) that covers this syntax.

File: after/syntax/tsx.vim (hunk 1 above)
Comment (dwalsh01) [reply after/syntax/tsx.vim]: The above revisions have been completed.

**Review by jparise — APPROVED** [review]

**Comment by jparise** [comment]
Thanks very much!

**Comment by kaicataldo** [comment]
Thanks for working on this!

**Comment by kaicataldo** [comment]
This unfortunately doesn't seem to work for .jsx files for me. 

I don't think it's a good idea to set
```
autocmd BufNewFile,BufRead *.tsx,*.jsx set filetype=typescript.tsx
```
for JavaScript files since I have other plugins that run on `.js{x?}` files. I'll see if I can find some time to also make this work for non-`.tsx` files out of the box.

**Comment by dwalsh01** [comment]
@kaicataldo I comment out the `autocmd` line in my files and still get the syntax highlighting in `.jsx` files. 

Here is a link to my `init.vim` should you wish to view it:

https://github.com/dwalsh01/vim-setup

**Pull request merged**